### PR TITLE
Elrondgo mandos exporter

### PR DESCRIPTION
**Pull request by andrei-diaconescu**

### CODE REVIEW

File: arwenmandos/util.go [hunk 1]
```diff
@@ -155,6 +158,45 @@ func addESDTToVMInput(esdtData []*mj.ESDTTxData, vmInput *vmcommon.VMInput) {
 	}
 }
 
+// CreateMultiTransferData builds data for a multiTransferESDT
+func CreateMultiTransferData(to []byte, esdtData []*mj.ESDTTxData, endpointName string, arguments [][]byte) []byte {
```
Comment (andrei-marinica) [inline arwenmandos/util.go:162]: also move this entire file and the tests to mandos-go/util, other projects will also make use of it

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

File: mandos-go/elrondgo-exporter/elrondGoExporter.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+		case *mj.TxStep:
+			if step.ExpectedResult.Status.Value.Cmp(okStatus) == 0 {
+
+				if step.Tx.GasPrice.Value == 0 {
+					step.Tx.GasPrice.Value = minimumAcceptedGasPrice
+				}
+				arguments := getArguments(step.Tx.Arguments)
+				switch step.StepTypeName() {
+				case "scCall":
+					if txIdRequieresBenchmark(step.TxIdent) && benchmarkTxPosIsNotSet(benchmarkTxPos) {
```
Comment (camilbancioiu) [inline mandos-go/elrondgo-exporter/elrondGoExporter.go:86]: Minor typo: `Requires`

File: mandos-go/elrondgo-exporter/elrondGoExporter.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+	txs = make([]*Transaction, 0)
+	deployTxs = make([]*Transaction, 0)
+	accounts = make([]*TestAccount, 0)
+	deployedAccounts = make([]*TestAccount, 0)
+
+	for i := 0; i < len(steps); i++ {
+		switch step := steps[i].(type) {
+		case *mj.SetStateStep:
+
```
Comment (camilbancioiu) [inline mandos-go/elrondgo-exporter/elrondGoExporter.go:69]: Delete empty line.

File: mandos-go/elrondgo-exporter/elrondGoExporter.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+	scenario, err := getScenario(mandosTestPath)
+	if err != nil {
+		return nil, nil, nil, nil, -1, err
+	}
+	steps := scenario.Steps
+	accounts, deployedAccounts, txs, deployTxs, benchmarkTxPos, err = getAccountsAndTransactionsFromSteps(steps)
+	if err != nil {
+		return nil, nil, nil, nil, -1, err
+	}
+	return accounts, deployedAccounts, txs, deployTxs, benchmarkTxPos, nil
```
Comment (camilbancioiu) [inline mandos-go/elrondgo-exporter/elrondGoExporter.go:40]: Optional, but strongly recommended: create a `struct` called `ScenarioWithBenchmark` (or similarly named) which contains all these values as fields.

**Review by camilbancioiu — CHANGES_REQUESTED** [review]

File: mandos-go/elrondgo-exporter/elrondGoExporter.go [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+		scAddress := append(ScAddressPrefix, newMandosAddressMock.NewAddress.Value[ScAddressPrefixLength:]...)
+		ownerAddress := newMandosAddressMock.CreatorAddress.Value
+		account := SetNewAccount(0, scAddress, big.NewInt(0), make(map[string][]byte), make([]byte, 0), ownerAddress)
+		deployedAccounts = append(deployedAccounts, account)
+	}
+
+	return accounts, deployedAccounts, nil
+}
+
+func convertMandosToTestAccount(mandosAcc *mj.Account) (*TestAccount, error) {
```
Comment (camilbancioiu) [inline mandos-go/elrondgo-exporter/elrondGoExporter.go:158]: Consider renaming to `convertMandosAccountToTestAccount`.

File: mandos-go/elrondgo-exporter/exportedTransaction.go [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+	return tx
+}
+
+func (tx *Transaction) GetCallArguments() [][]byte {
+	return tx.args
+}
+
+func (tx *Transaction) WithSenderAddress(address []byte) *Transaction {
+	tx.sndAddr = make([]byte, 0)
+	tx.sndAddr = address
```
Comment (camilbancioiu) [inline mandos-go/elrondgo-exporter/exportedTransaction.go:90]: You probably wanted this:
```suggestion
	tx.sndAddr = make([]byte, len(address))
	tx.sndAddr = copy(tx.sndAddr, address)
```

**Review by andrei-marinica — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Pull request merged**